external/esp_idf_port/esp32/esp32_wifi_utils.c

```diff
@@ -0,0 +1,740 @@
+			switch (ap_list_buffer[i].authmode) {
+			case WIFI_AUTH_OPEN:
+				scan_list[i].ap_info.ap_auth_type = WIFI_UTILS_AUTH_OPEN;
+				break;
+			case WIFI_AUTH_WEP:
+				scan_list[i].ap_info.ap_auth_type = WIFI_UTILS_AUTH_WEP_SHARED;
+				break;
+			case WIFI_AUTH_WPA_PSK:
+				scan_list[i].ap_info.ap_auth_type = WIFI_UTILS_AUTH_WPA_PSK;
+				break;
+			case WIFI_AUTH_WPA2_PSK:
+				scan_list[i].ap_info.ap_auth_type = WIFI_UTILS_AUTH_WPA2_PSK;
+				break;
+			case WIFI_AUTH_WPA_WPA2_PSK:
+				scan_list[i].ap_info.ap_auth_type = WIFI_UTILS_AUTH_WPA_AND_WPA2_PSK;
+				break;
+			case WIFI_AUTH_WPA2_ENTERPRISE:
+			default:
+				scan_list[i].ap_info.ap_auth_type = WIFI_UTILS_AUTH_UNKNOWN;
+				break;
+			}
+
+			switch (ap_list_buffer[i].pairwise_cipher) {
+			case WIFI_CIPHER_TYPE_NONE:
+				scan_list[i].ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_NONE;
+				break;
+			case WIFI_CIPHER_TYPE_WEP40:
+				scan_list[i].ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_WEP_64;
+				break;
+			case WIFI_CIPHER_TYPE_WEP104:
+				scan_list[i].ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_WEP_128;
+				break;
+			case WIFI_CIPHER_TYPE_TKIP:
+				scan_list[i].ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP;
+				break;
+			case WIFI_CIPHER_TYPE_CCMP:
+				scan_list[i].ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_AES;
+				break;
+			case WIFI_CIPHER_TYPE_TKIP_CCMP:
+				scan_list[i].ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP_AND_AES;
+				break;
+			default:
+				scan_list[i].ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_UNKNOWN;
+				break;
+			}
+
+			if (i > 0) {
+				scan_list[i - 1].next = &scan_list[i];
+			}
+		}
+		g_cbk.scan_done(WIFI_UTILS_SUCCESS, scan_list, NULL);
+	}
+	free(ap_list_buffer);
+	free(scan_list);
+}
+
+static esp_err_t event_handler(void *ctx, system_event_t *event)
+{
+	switch (event->event_id) {
+	case SYSTEM_EVENT_SCAN_DONE: {
+		scan_done_handler();
+		ESP_LOGI(TAG, "SYSTEM_EVENT_SCAN_DONE ");
+		break;
+	}
+	case SYSTEM_EVENT_STA_START: {
+		isStaConnected = 0;
+		ESP_LOGI(TAG, "sta start");
+		break;
+	}
+	case SYSTEM_EVENT_STA_STOP: {
+		isStaConnected = 0;
+		ESP_LOGI(TAG, "sta stop");
+		break;
+	}
+	case SYSTEM_EVENT_STA_GOT_IP: {
+		ESP_LOGI(TAG, "got ip:%s", ip4addr_ntoa(&event->event_info.got_ip.ip_info.ip));
+		break;
+	}
+	case SYSTEM_EVENT_STA_LOST_IP: {
+		ESP_LOGI(TAG, "[%s] lost ip", __func__);
+		break;
+	}
+	case SYSTEM_EVENT_STA_CONNECTED: {
+		s_retry_num = 0;
+		isStaConnected = 1;
+		g_cbk.sta_connected(WIFI_UTILS_SUCCESS, NULL);
+		ESP_LOGI(TAG, "[%s] SYSTEM_EVENT_STA_CONNECTED", __func__);
+		break;
+	}
+	case SYSTEM_EVENT_STA_DISCONNECTED: {
+		s_retry_num++;
+		int reason = event->event_info.disconnected.reason;
+		ESP_LOGI(TAG, "[%s] connect to the AP %s fail %d: %d; %d\n", __func__, event->event_info.disconnected.ssid, s_retry_num, event->event_info.disconnected.reason, isStaConnected);
+
+		if (s_retry_num > 0) {
+
+			printf("g_cbk.sta_connected fail\n");
```

Comment:
1. blank line,
2. printf in this file, replace with debug log

Reply:
right, done